src/pages/ProductsPage/Products.tsx

```diff
@@ -1,36 +1,91 @@
 import "./products.scss";
-import React, { useState, useEffect } from "react";
+import arrow from "../../assets/Mask.svg";
+import { useEffect } from "react";
 import Product from "../../components/ui/productCard/ProductCard";
 import Loader from "../../components/ui/loader/Loader";
-import { BreedType } from "../../types";
+import { useFetchBreedsQuery } from "../../components/ui/redux-paginstion-slice/fetch-slice";
+import { usePagination, useAppDispatch } from "../../shared/utils";
+import { changePage } from "../../components/ui/redux-paginstion-slice/fetch-slice";
 
 export default function Products() {
-  const [breeds, setBreeds] = useState<BreedType[]>([]);
+  const dispatch = useAppDispatch();
+  const numberOfCards = 10;
+  const {
+    firstContentIndex,
+    lastContentIndex,
+    nextPage,
+    prevPage,
+    page,
+    setPage,
+    totalPages,
+  } = usePagination({
+    numberOfCards: numberOfCards,
+    listLength: 170,
+  });
+  const { data = [] } = useFetchBreedsQuery(page * numberOfCards);
+  let breeds = data;
   useEffect(() => {
-    fetch("https://api.thedogapi.com/v1/breeds?limit=25&page=0")
-      .then((res) => res.json())
-      .then((data) => setBreeds(data))
-      .catch((error) => {
-        console.error("Error fetching breed information:", error);
-      });
-  }, [breeds]);
+    window.scrollTo({
+      top: 0,
+      behavior: "smooth",
+    });
+    dispatch(changePage(page));
+  }, [dispatch, page]);
   return (
     <div className="products container-fluid">
       <h1 className="products__title">product page</h1>
-      {breeds.length ? (
+      {breeds.length >= page * 10 ? (
         <div className="row products-page">
-          {breeds.map((dogy, index) => (
-            <div
-              key={dogy.id}
-              className="col-12 col-md-6 col-lg-3 products-page__card"
-            >
-              {dogy && <Product {...dogy} />}
-            </div>
-          ))}
+          {breeds
+            .slice(firstContentIndex, lastContentIndex)
+            .map((dogy, index) => (
+              <div
+                key={dogy.id}
+                className="col-12 col-md-6 col-lg-3 products-page_card"
+              >
+                {dogy && <Product {...dogy} />}
+              </div>
+            ))}
         </div>
       ) : (
         <Loader />
       )}
+      <div className="products-pagination">
+        <button onClick={prevPage} className="products-pagination__page-button">
+          <img
+            alt="slider__arrow"
+            className="products-pagination__page-button-arrow"
+            src={arrow}
+          />
+        </button>
+        {Array.from({ length: totalPages }, (_, i) => i).map((el) => (
```

Comment:
consider moving this code to the separate UI component called Pagination (it's better to use third-party package like https://www.npmjs.com/package/react-paginate instead)

at this moment the code is quite hard to read in the context if the Products component